pkg/tfbridge/provider.go

```diff
@@ -348,8 +339,11 @@ func (p *Provider) CheckConfig(ctx context.Context, req *pulumirpc.CheckRequest)
 		return nil, validationErrors
 	}
 
+	// Ensure propreties marked secret in the schema have secret values.
+	secretNews := MarkSchemaSecrets(ctx, p.config, p.info.Config, resource.NewObjectProperty(news)).ObjectValue()
```

Comment:
MarkSchemaSecrets takes over from ConfigEncoding to ensure the provider marks nested secrets.